sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/RemovableSymbolCollector.cs

```diff
@@ -0,0 +1,189 @@
+        }
+
+        public override void VisitStructDeclaration(StructDeclarationSyntax node)
+        {
+            ConditionalStore(GetDeclaredSymbol(node), IsRemovableType);
+            base.VisitStructDeclaration(node);
+        }
+
+        public override void VisitInterfaceDeclaration(InterfaceDeclarationSyntax node)
+        {
+            ConditionalStore(GetDeclaredSymbol(node), IsRemovableType);
+            base.VisitInterfaceDeclaration(node);
+        }
+
+        public override void VisitDelegateDeclaration(DelegateDeclarationSyntax node)
+        {
+            ConditionalStore(GetDeclaredSymbol(node), IsRemovableType);
+            base.VisitDelegateDeclaration(node);
+        }
+
+        public override void VisitFieldDeclaration(FieldDeclarationSyntax node)
+        {
+            StoreRemovableVariableDeclarations(node);
+            base.VisitFieldDeclaration(node);
+        }
+
+        public override void VisitEventFieldDeclaration(EventFieldDeclarationSyntax node)
+        {
+            StoreRemovableVariableDeclarations(node);
+            base.VisitEventFieldDeclaration(node);
+        }
+
+        public override void VisitEventDeclaration(EventDeclarationSyntax node)
+        {
+            ConditionalStore(GetDeclaredSymbol(node), IsRemovableMember);
+            base.VisitEventDeclaration(node);
+        }
+
+        public override void VisitPropertyDeclaration(PropertyDeclarationSyntax node)
+        {
+            ConditionalStore(GetDeclaredSymbol(node), IsRemovableMember);
+            base.VisitPropertyDeclaration(node);
+        }
+
+        public override void VisitIndexerDeclaration(IndexerDeclarationSyntax node)
+        {
+            ConditionalStore(GetDeclaredSymbol(node), IsRemovableMember);
+            base.VisitIndexerDeclaration(node);
+        }
+
+        public override void VisitMethodDeclaration(MethodDeclarationSyntax node)
+        {
+            var symbol = (IMethodSymbol)GetDeclaredSymbol(node);
+            ConditionalStore(symbol.PartialDefinitionPart ?? symbol, IsRemovableMethod);
+            base.VisitMethodDeclaration(node);
+        }
+
+        public override void VisitConstructorDeclaration(ConstructorDeclarationSyntax node)
+        {
+            if (!IsEmptyConstructor(node))
+            {
+                ConditionalStore((IMethodSymbol)GetDeclaredSymbol(node), IsRemovableMethod);
+            }
+            base.VisitConstructorDeclaration(node);
+        }
+
+        private bool IsRemovableType(ISymbol typeSymbol)
+        {
+            var accessibility = typeSymbol.GetEffectiveAccessibility();
+            return typeSymbol.ContainingType != null
+                && (accessibility == Accessibility.Private || accessibility == Accessibility.Internal)
+                && IsRemovable(typeSymbol);
+        }
+
+        private bool IsRemovableMember(ISymbol symbol) =>
+            symbol.GetEffectiveAccessibility() == Accessibility.Private &&
```

Comment:
Does this mean that internal properties, fields and events will only be considered removable? If so, why are they treated differently from types and methods?